irida_import/irida_import.py

```diff
@@ -628,6 +648,68 @@ def link(self, sample_file, folder_id):
 
         return added
 
+    def upload_file_to_galaxy(self, sample_file, folder_id):
+        """
+        Upload a sample file to Galaxy
+
+        :type sample_file: SampleFile
+        :param sample_file: the sample file to upload
+        :type folder_id: ID of folder to upload file to
+        :param sample_folder_path: the folder in Galaxy to store the file in
+        :return: a list containing a single dict with the file's
+        url, id, and name.
+        """
+        self.logger.debug('Attempting to upload file')
+        added = None
+        file_path = sample_file.path
+        self.logger.debug(
+            "       Sample file's local path is" + file_path)
+        file_type = 'auto'
+        file_ext = os.path.splitext(file_path)[1]
+
+        # Assume fastq files are fastqsanger:
+        if file_ext == '.fastq':
+            file_type = 'fastqsanger'
+
+        tmp_dir = tempfile.mkdtemp()
+        tmp_file_mode = 'w+b'
+
+        try:
+            tmp_file = tempfile.NamedTemporaryFile(mode=tmp_file_mode, prefix=sample_file.name, dir=tmp_dir)
+            tmp_file.name = tmp_dir + "/" + sample_file.name
+
+            if file_ext == ".fasta":
```

Comment:
do we need to handle fast5 data here also? Or is that not yet supported in this tool

Reply:
Currently fast5 works exactly how the .fastq files works by sending an accept header of application/fastq. We will need to update IRIDA before we can update the tool to send the correct accept header. This will potentially break functionality for older versions of IRIDA however